float32[] voltage_input
